src/js/core.js

```diff
@@ -945,6 +949,37 @@ function MediumEditor(elements, options) {
                 }
             }
 
+            // If the selection is right at the ending edge of a link, put it outside the anchor tag instead of inside.
```

Comment:
Yeah, I agree that we wouldn't want anything core calling out into an extension which may or may not be there.  An alternative would be to have the extension override the 'importSelection' method and catch all calls to it, but I think that seems a bit unnecessary.

I think a separate helper function is a good idea for now, maybe we can improve on this and try to make it more and more generic and reusable over time.